Server side sell page

tendergorilla55 opened the pull request:
## WHAT
商品登録のサーバーサイド作成

## WHY
出品する商品の登録をできるようにするため

tendergorilla55 committed:
Edit mark up

写真の選択フォルダを表示するように変更

tendergorilla55 committed:
Create sell method

カテゴリーにreference item_idを追加
Createページの作成
Createページのscss追加

tendergorilla55 committed:
Merge remote-tracking branch 'Kentarou-Orido/master' into server-side-sell-page

tendergorilla55 committed:
Spell miss

スペルミス修正

tendergorilla55 committed:
Merge branch 'master' of https://github.com/Kentarou-Orido/freemarket_sample_37e into server-side-sell-page

# Conflicts:
#	config/routes.rb

tendergorilla55 committed:
Add pry-rails

テスト用にpry-railsのgemを追加

tendergorilla55 committed:
Merge remote-tracking branch 'Kentarou-Orido/master' into server-side-sell-page

# Conflicts:
#	db/schema.rb

tendergorilla55 committed:
Edit image folder

トップ画像が表示されるように修正

tendergorilla55 committed:
Merge remote-tracking branch 'Kentarou-Orido/master' into server-side-sell-page

# Conflicts:
#	Gemfile
#	Gemfile.lock
#	config/routes.rb
#	db/schema.rb

tendergorilla55 committed:
Edit schema

updateミスの修正

tendergorilla55 committed:
Edit column name

DBテーブル名に合わせて修正

tendergorilla55 committed:
Edit spellmiss

スペルミス修正

tendergorilla55 committed:
Merge remote-tracking branch 'Kentarou-Orido/master' into server-side-sell-page

# Conflicts:
#	app/models/category.rb

tendergorilla55 committed:
Edit files

Seedファイルを一部修正
Paramsをカラムと統一
uploaderの追加
faker, carrier wave, mini_magickの追加
SelectのデータをDBから抽出
CSVデータを一部修正

tendergorilla55 committed:
Implementing functions

画像の送信実装
カテゴリーグループの保存実装

tendergorilla55 committed:
Merge remote-tracking branch 'Kentarou-Orido/master' into server-side-sell-page

# Conflicts:
#	app/models/item.rb
#	db/schema.rb

tendergorilla55 committed:
Add brand function

ブランドを追加できるように変更

tendergorilla55 committed:
Make selectbox.js

配送料負担を選択すると配送の方法が出てくる

tendergorilla55 committed:
Add price.js

金額を入力すると、手数料と販売利益が変更される

tendergorilla55 committed:
edit image piece

枚数変更

tendergorilla55 committed:
Edit test

テストコードの編集

TOMARUMARU commented on app/assets/javascripts/price.js:
```diff
@@ -0,0 +1,19 @@
+
+  $(".input-price").on("keyup", function(){
+    var price = parseInt($(this).val());
+    if(300 <= price && price < 10000000){
```
```300 <= price < 10000000```
で良さそうです

tendergorilla55 replied:
そうですね。修正しました！

TOMARUMARU commented on app/assets/javascripts/selectbox.js:
```diff
@@ -0,0 +1,28 @@
+    }
+  });
+
+  // $(".select-category").on("change", function() {
```
ここら辺のコメントアウトは使用予定ですか？
使用する予定があるならコメントを残し、なければ削除しましょう

tendergorilla55 replied:
カテゴリーのセレクタを作るときに必要になりそうなので残しておきます。

TOMARUMARU commented on app/controllers/sells_controller.rb:
```diff
@@ -1,6 +1,19 @@
+
+  def create
+    @item = Item.new(sell_params)
+    @item.save
```
saveできなかった時の処理は不要ですか？

tendergorilla55 replied:
必要なので、追加しました。

TOMARUMARU commented on app/views/sells/create.html.haml:
```diff
@@ -0,0 +1,8 @@
+    %h3.modal-head 出品が完了しました
+    .sell-modal-body
+      あなたが出品した商品は「出品した商品一覧」からいつでも見ることができます。
+      = link_to '続けて出品する', '/sell', class: 'btn-default btn-red'
```
prefixで書きましょう

tendergorilla55 replied:
了解です。修正しました。

TOMARUMARU commented on app/views/sells/create.html.haml:
```diff
@@ -0,0 +1,8 @@
+      あなたが出品した商品は「出品した商品一覧」からいつでも見ることができます。
+      = link_to '続けて出品する', '/sell', class: 'btn-default btn-red'
+      %p.text-center
+        =link_to '商品ページへいってシェアする', '/items',
```
prefixで書きましょう

tendergorilla55 replied:
同じく修正しました。

TOMARUMARU commented on db/migrate/20190220062736_add_categories_to_category_ids.rb:
```diff
@@ -1,5 +1,5 @@
 class AddCategoriesToCategoryIds < ActiveRecord::Migration[5.0]
   def change
-    add_reference :categories, :item, null:false
+    # add_reference :categories, :item, null:false
```
コメントアウトしたmigrationファイルはあまり残したくないですね...

tendergorilla55 committed:
コメントに対する修正

price.jsを簡素化
Selectbox.jsのコメントアウトの説明追加
Prefixでの表記に変更
出品ボタンのリンク作成
@item.saveをif文で分岐

tendergorilla55 committed:
Merge remote-tracking branch 'Kentarou-Orido/master' into server-side-sell-page

tendergorilla55 committed:
アラーム追加

If文での分岐を追加
アラームのflashを追加

TOMARUMARU reviewed (approved):
LGTM

tendergorilla55 committed:
Add validate

Sizeが抜けてたので追加

tendergorilla55 committed:
current_userに変更

ログイン後のことを考えてcurrent_user.idに変更

Kentarou-Orido committed:
Merge branch 'master' into server-side-sell-page
